ingenialink/servo.py

```diff
@@ -41,6 +47,87 @@
 OPERATION_TIME_OUT = -3
 
 
+class DictionaryFactory:
+    """Dictionary factory, creates a dictionary instance depends on the file version
+    and connection interface"""
+
+    _VERSION_ABSOLUTE_PATH = "Header/Version"
+    _VERSION_REGEX = r"(\d+)\.*(\d*)"
+    _MAJOR_VERSION_GROUP = 1
+    _MINOR_VERSION_GROUP = 2
+
+    @classmethod
+    def create_dictionary(cls, dictionary_path: str, interface: Interface) -> Dictionary:
+        """Creates a dictionary instance choosing the class depending on dictionary version and
+         connection interface.
+
+        Args:
+            dictionary_path: target dictionary path
+            interface: connection interface
+
+        Returns:
+            Dictionary instance
+
+        Raises:
+            FileNotFoundError: dictionary path does not exist.
+            ILDictionaryParseError: xdf is not well-formed.
+            ILDictionaryParseError: File is not a xdf.
+            NotImplementedError: Dictionary version is not supported.
+
+        """
+        major_version, minor_version = cls.__get_dictionary_version(dictionary_path)
+        if major_version == 3:
+            return DictionaryV3(dictionary_path, interface)
+        if major_version == 2:
+            if interface == Interface.CAN:
+                return CanopenDictionaryV2(dictionary_path)
+            if interface == Interface.ECAT:
+                return EthercatDictionaryV2(dictionary_path)
+            if interface in [Interface.ETH, Interface.EoE]:
+                return EthernetDictionaryV2(dictionary_path)
+            if interface == Interface.VIRTUAL:
+                return VirtualDictionary(dictionary_path)
+        raise NotImplementedError(f"Dictionary version {major_version} is not supported")
+
+    @classmethod
+    def __get_dictionary_version(cls, dictionary_path: str) -> Tuple[int, int]:
+        """Return dictionary version, major and minor version.
+
+        Args:
+            dictionary_path: dictionary path
+
+        Returns:
+            Major and minor version
+
+        Raises:
+            FileNotFoundError: dictionary path does not exist.
+            ILDictionaryParseError: xdf is not well-formed
+            ILDictionaryParseError: File is not a xdf
+
+        """
+        try:
```

Comment:
This try/except block is too broad. I would check if the file exists and if it does not raise the FileNotFoundError